api/v1beta1/machine_types.go

```diff
@@ -143,6 +168,13 @@ type MachineSpec struct {
 	NodeDeletionTimeout *metav1.Duration `json:"nodeDeletionTimeout,omitempty"`
 }
 
+// MachineReadinessGate contains the type of a Machine condition to be used as a readiness gate.
+type MachineReadinessGate struct {
+	// conditionType refers to a condition with matching type in the Machine's condition list.
+	// Note: Both Cluster API conditions or conditions added by 3rd party controllers can be used as readiness gates.
+	ConditionType string `json:"conditionType"`
```

Comment:
Should mark the field explicitly as required

Should also set min/max lengths on the field

Have we considered making these enums? Since we do have control over the condition types?

Reply:
These can also be third party conditions, so we don't know all of them

Reply:
Added validation, we can't make it an enum